IdentityCore/src/workplacejoin/MSIDWorkPlaceJoinUtilBase.m

```diff
@@ -147,12 +152,20 @@ For asymmetric keys this holds the public key hash which allows digital identity
     }
     
     NSMutableDictionary *mutableCertQuery = [NSMutableDictionary new];
-    
     if (certAttributes)
     {
         [mutableCertQuery addEntriesFromDictionary:certAttributes];
     }
     
+#if TARGET_OS_OSX
+    // For macOS, if the key is ECC key, use shared access group to query certificate. If it is RSA, remove shared access group from query for certificate as it is not login keychain.
```

Comment:
Actually related question to Hieu's question. Do you need to set to use data protection keychain somewhere for metadata in the query for ECC + macOS?

Reply:
Noting our offline discussions here,
Since tenantId is the only 'metadata' required, for default registrations querying metadata is not necessary as it is tied to the primary key. ECC registration components are iOS style keychain items and if querying metadata related to them, we would need to set [kSecUseDataProtectionKeychain](https://developer.apple.com/documentation/security/ksecusedataprotectionkeychain?language=objc) to true but we are not querying metadata for ECC registrations as it is not required.
However for legacy registrations, we need to perform a check to match tenantId in metadata matches the one client received in PKeyAuth challenge. By default [kSecUseDataProtectionKeychain](https://developer.apple.com/documentation/security/ksecusedataprotectionkeychain?language=objc) is set to NO and should not be set to YES for legacy keychain items as they do not behave like iOS style keychain items.